### DISCUSSION TITLE
Initialize State at app startup

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#9923] [CLI - Electron mode] [answered]
Where or how do I best initialize the state of my app at a startup?
My state.js file looks like this...

You'll see in the commented lines i need to set the apps Dark theme and primary color based on values in the app's state, which will eventually be reading the users saved settings so they can choose the app is light or dark theme. Should i be doing the commented lines in my store/index.js file?

**store/app/state.js**
```
import { Dark } from 'quasar';
import { getCssVar, setCssVar } from 'quasar'

export default function () {
  return {
    // app
    isBusy: false,
    isDarkThemeEnabled: false,
    isCategoriesEnabled: true,
    primaryColor: '',
    thumbnailSize: 64,
    configFilepath: '',
  }
}

// init
// Dark.set(state.isDarkThemeEnabled);
// setCssVar('primary', state.primaryColor)
```

**store/index.js**
```
import { store } from 'quasar/wrappers'
import { createStore } from 'vuex'

import app from './app';

/*
 * If not building with SSR mode, you can
 * directly export the Store instantiation;
 *
 * The function below can be async too; either use
 * async/await or return a Promise which resolves
 * with the Store instance.
 */

export default store(function (/* { ssrContext } */) {
  const Store = createStore({
    modules: {
      app
    },

    // enable strict mode (adds overhead!)
    // for dev mode and --debug builds only
    strict: process.env.DEBUGGING
  })

  return Store
})

```

**Comment by ontwikkelfabriek** [comment, marked as answer]
@JokerMartini 

The vuex store is a good place to put your user settings, except it does not persist data by default. So you'll loose all the settings the user changed after a refresh or a new session.  You can either use something like vuex-persist( vuex plugin) or populate the store from localstorage ( use the quasar plugin, also you'll have to set the values in localstorage when the user changes settings). You can use a boot file with access to the store to run your `init`.

**Reply by JokerMartini** [reply]
So that is what I'm kind of doing but do where do I put the logic that reads from local storage and sets the state of the store? Do I do that in the boot init file itself or have the init file call an action that loads from local storage and sets the state?

**Reply by ontwikkelfabriek** [reply]
@JokerMartini 

You start in a boot file, It's up to you if you put the code of the `logic that reads from localstorage` explicitly in the boot file ( works perfectly fine ,but it's probably not the most _clean_ way) or you call your `init` function from the boot file.
